shaccson/src/main/java/UserFormServlet.java

```diff
@@ -0,0 +1,58 @@
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/user/form")
+public class UserFormServlet extends HttpServlet {
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String idParam = request.getParameter("id");
+
+        if (idParam != null) {
+            int id = Integer.parseInt(idParam);
+            UserDao userDao = new UserDao();
+            User user = userDao.getUserById(id);
+            request.setAttribute("user", user);
+        }
+
+        getServletContext().getRequestDispatcher("/users/form.jsp")
+                .forward(request, response);
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        UserDao userDao = new UserDao();
+        String idParam = request.getParameter("id");
+        String username = request.getParameter("username");
+        String email = request.getParameter("email");
+        String password = request.getParameter("password");
+
+        User user = new User();
```

Comment:
CLEAN-CODE: pouzivat nemenne promenne vsude kde to je mozne v aplikaci.

```suggestion
final User user = new User();
```

**Duvod**

Používání neměnných proměnných final v Javě je dobré, protože zajišťuje, že hodnota proměnné se po jejím přiřazení nemůže změnit, což zvyšuje čitelnost a spolehlivost kódu. Pomáhá to také při optimalizaci kódu, protože kompilátor může lépe předvídat chování programu. Dále zajišťuje, že nebudou neúmyslné změny hodnoty proměnných, což snižuje riziko chyb, a také to usnadňuje práci s multithreadingem, kde je důležité, aby hodnoty nebyly modifikovány souběžně v různých vláknech.

Reply:
pouzity nemenne promenne